app/controllers/vm_common.rb

```diff
@@ -1204,11 +1204,12 @@ def set_right_cell_vars(options = {})
       }
       action = "remove_security_group"
     when "clone", "migrate", "publish"
+      @record ||= identify_record(params[:prov_id], VmOrTemplate)
```

Comment:
NOTE I'm unsure where `@record` is usually set, but without this the `model_for_vm(@record)` won't work because `@record` is `nil`.

@record is used below in the `when "edit"` case for the same `model_for_vm(@record).display_name`

Reply:
I did check that this works with VM clone, migrate, and publish

Reply:
@jeffibm do you know where `@record` gets set normally?  Should I do this in the `vm_clone` / `vm_publish` methods?

Reply:
Okay setting it in `vm_clone` / `vm_migrate` / `vm_publish` feels like less of a hack because it is similar to other methods in that controller